tests/util/src/opentelemetry/test/mock_textmap.py

```diff
@@ -60,39 +53,31 @@ class MockTextMapPropagator(TextMapPropagator):
 
     def extract(
         self,
-        getter: Getter[TextMapPropagatorT],
-        carrier: TextMapPropagatorT,
+        carrier: typing.Dict[str, str],
         context: typing.Optional[Context] = None,
     ) -> Context:
-        trace_id_list = getter.get(carrier, self.TRACE_ID_KEY)
-        span_id_list = getter.get(carrier, self.SPAN_ID_KEY)
+        trace_id = carrier.get(self.TRACE_ID_KEY)
+        span_id = carrier.get(self.SPAN_ID_KEY)
```

Comment:
Do we expect the values to be single items instead of lists now? It is possible that a carrier is a dict like object but has a collection as value, for example, to represent multiple HTTP headers with same name. We might want to support both cases here. It should be relatively cheap now that we consolidated this logic in a single place.